src/fondant/compiler.py

```diff
@@ -183,3 +184,61 @@ def _generate_spec(self, pipeline: Pipeline, extra_volumes: list) -> dict:
             "version": "3.8",
             "services": services,
         }
+
+
+class KubeFlowCompiler(Compiler):
+    """Compiler that creates a Kubeflow pipeline spec from a pipeline."""
+
+    def _resolve_imports(self):
+        """Resolve imports for the Kubeflow compiler."""
+        try:
+            import kfp
+            from kfp import dsl
+        except ImportError:
+            raise ImportError(
+                "You need to install kfp to use the Kubeflow compiler, "
+                / "you can install it with `poetry install --extras kfp`"
```

Comment:
```suggestion
                / "you can install it with `pip install fondant[kfp]`"
```

`poetry install` is only for fondant developers. Users can also use poetry, but then they'll install their own project and need to indicate the fondant extras with the fondant dependency in their `pyproject.toml`.

Reply:
https://github.com/ml6team/fondant/blob/86867a5e777c93a7ca0d982f263d66c38abc3e8f/src/fondant/import_utils.py#L24C1-L25C1